sdk/webpubsub/azure-messaging-webpubsub/src/test/java/com/azure/messaging/webpubsub/WebPubSubServiceAsyncClientTests.java

```diff
@@ -0,0 +1,314 @@
+            requestOptions));
+
+        assertResponse(client.sendToAllWithResponse(BinaryData.fromString("{\"intvalue\": 25}"),
+            requestOptions));
+
+        assertResponse(client.sendToAllWithResponse(BinaryData.fromString("{\"floatvalue\": 55.4}"),
+            requestOptions));
+    }
+
+    @Test
+    public void testRemoveNonExistentUserFromHub() {
+        StepVerifier.create(client.removeUserFromAllGroupsWithResponse("testRemoveNonExistentUserFromHub", new RequestOptions()))
+            .assertNext(response -> {
+                assertEquals(204, response.getStatusCode());
+            })
+            .expectComplete()
+            .verify(TIMEOUT);
+    }
+
+    @Test
+    public void testRemoveConnectionFromAllGroup() {
+        StepVerifier.create(client.removeConnectionFromAllGroupsWithResponse("test_connection", new RequestOptions()))
+            .assertNext(response -> {
+                assertEquals(204, response.getStatusCode());
+            })
+            .expectComplete()
+            .verify(TIMEOUT);
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void testGetAuthenticationToken() {
+        StepVerifier.create(client.getClientAccessToken(new GetClientAccessTokenOptions()))
+            .assertNext(token -> {
+                Assertions.assertNotNull(token);
+                Assertions.assertNotNull(token.getToken());
+                Assertions.assertNotNull(token.getUrl());
+
+                assertTrue(token.getUrl().startsWith("wss://"));
+                assertTrue(token.getUrl().contains(".webpubsub.azure.com/client/hubs/"));
+
+                String authToken = token.getToken();
+                JWT jwt;
+                try {
+                    jwt = JWTParser.parse(authToken);
+                } catch (ParseException e) {
+                    fail("Unable to parse auth token: " + authToken + " exception: ", e);
+                    return;
+                }
+
+                JWTClaimsSet claimsSet;
+                try {
+                    claimsSet = jwt.getJWTClaimsSet();
+                } catch (ParseException e) {
+                    fail("Unable to parse claims: " + authToken + " exception: ", e);
+                    return;
+                }
+
+                assertNotNull(claimsSet);
+                assertNotNull(claimsSet.getAudience());
+                assertFalse(claimsSet.getAudience().isEmpty());
+
+                String aud = claimsSet.getAudience().iterator().next();
+                assertTrue(aud.contains(".webpubsub.azure.com/client/hubs/"));
+            })
+            .expectComplete()
+            .verify(TIMEOUT);
+
+    }
+
+    /*****************************************************************************************************************
+     * Sync Tests - WebPubSubGroup
+     ****************************************************************************************************************/
```

Comment:
We should separate these out to its respective client test file.

Reply:
This is leftover. I will remove it.